workflow/controller/operator.go

```diff
@@ -3334,3 +3422,19 @@ func (woc *wfOperationCtx) substituteGlobalVariables() error {
 	}
 	return nil
 }
+
+func (woc *wfOperationCtx) DeleteAgentPod(ctx context.Context) error {
+	agentPodName, ok := woc.wf.Labels[common.LabelKeyAgentPod]
```

Comment:
you don't need this label (and any bugs it'll create) because `podName := woc.wf.NodeID("agent") + "-agent"`